modules/db_instance/variables.tf

```diff
@@ -180,7 +180,7 @@ variable "multi_az" {
 variable "iops" {
   description = "The amount of provisioned IOPS. Setting this implies a storage_type of 'io1'"
   type        = number
-  default     = 0
+  default     = null
```

Comment:
This is a very candide proposal, I have no idea if there is a good reason or not for the default value currently been `0`.

The only thing I know: this is currently causing an incompatibility with the default behavior of the AWS Provider in his version 4.46.0.

Someone, not using `terraform-aws-rds` module, can directly transition from `gp2` to `gp3` (see: https://github.com/hashicorp/terraform-provider-aws/issues/28271#issuecomment-1348194124)

But using `terraform-aws-rds`, you are forced to override the default value of `iops` from `0`, to `null`:
https://github.com/hashicorp/terraform-provider-aws/issues/28271#issuecomment-1349076641

Reply:
its optional so `null` should be the appropriate default value